input/script_utils.go

```diff
@@ -726,6 +726,78 @@ func SecondLevelHtlcScript(revocationKey, delayKey *btcec.PublicKey,
 	return builder.Script()
 }
 
+// LeaseSecondLevelHtlcScript is the uniform script that's used as the output for
+// the second-level HTLC transactions. The second level transaction acts as a
+// sort of covenant, ensuring that a 2-of-2 multi-sig output can only be
+// spent in a particular way, and to a particular output.
+//
+// Possible Input Scripts:
+//  * To revoke an HTLC output that has been transitioned to the claim+delay
+//    state:
+//    * <revoke sig> 1
+//
+//  * To claim an HTLC output, either with a pre-image or due to a timeout:
+//    * <delay sig> 0
+//
+// OP_IF
+//     <revoke key>
+// OP_ELSE
+//     <lease maturity in blocks>
+//     OP_CHECKLOCKTIMEVERIFY
+//     OP_DROP
+//     <delay in blocks>
+//     OP_CHECKSEQUENCEVERIFY
+//     OP_DROP
+//     <delay key>
+// OP_ENDIF
+// OP_CHECKSIG
+func LeaseSecondLevelHtlcScript(revocationKey, delayKey *btcec.PublicKey,
+	csvDelay, cltvExpiry uint32) ([]byte, error) {
+
+	builder := txscript.NewScriptBuilder()
+
+	// If this is the revocation clause for this script is to be executed,
+	// the spender will push a 1, forcing us to hit the true clause of this
+	// if statement.
+	builder.AddOp(txscript.OP_IF)
+
+	// If this this is the revocation case, then we'll push the revocation
+	// public key on the stack.
+	builder.AddData(revocationKey.SerializeCompressed())
+
+	// Otherwise, this is either the sender or receiver of the HTLC
+	// attempting to claim the HTLC output.
+	builder.AddOp(txscript.OP_ELSE)
+
+	// The channel initiator always has the additional channel lease
+	// expiration constraint for outputs that pay to them which must be
+	// satisfied.
+	builder.AddInt64(int64(cltvExpiry))
```

Comment:
Wish we had something to reduce the duplication a bit here, given only these 3 lines are actually unique here....

On the otherhand, I like that things are fully explicit here. One alternative would be to make the `SecondLevelHtlcScript` as is into a private helper function, add a new param to let users splice in a custom directive, then make a new public `SecondLevelHtlcScript` function w/ a nil param, and specify it here. 

We'd only really use it once (so far), so I'd say it likely isn't worth it for now, as we don't know if this will really be the _right_ abstraction here.

Reply:
> We'd only really use it once (so far), so I'd say it likely isn't worth it for now, as we don't know if this will really be the right abstraction here.

Agreed, this is why I chose to explicitly copy them over.